packages/atlas/src/joystream-lib/extrinsics.ts

```diff
@@ -1163,4 +1163,47 @@ export class JoystreamLibExtrinsics {
 
     return { block }
   }
+
+  purchaseTokenOnMarketTx = async (tokenId: string, memberId: string, amount: string) => {
+    const amountCast = createType('u128', new BN(amount))
+
+    return this.api.tx.projectToken.buyOnAmm(
+      parseInt(tokenId),
+      parseInt(memberId),
+      amountCast,
+      createType('Option<ITuple<[Permill, u128]>>', [createType('Permill', new BN(0.5)), amountCast]) // percent, number of joy user wants to pay --- default on 0.5%
```

Comment:
I think 0.5 percent is 5000 permill (or is it 5 ? I'm never sure if `permill` stands for per 1M or 1K).
Also `new BN(0.5)` is `0` as bn.js doesn't support decimal

Reply:
Hmmm, in issue token transaction we use permill as a normal percentage value and it was parsed correctly by the Orion, plus In Orion tests Ignazio approached it the same. 

But definitely, BN from 0.5 will be a problem since the default value should be 0.5%, I hope `createType('Permill', 0.5)` will work